src/rabbit_mgmt_stats_gc.erl

```diff
@@ -134,15 +140,56 @@ gc_batch(Rows, Policies, State = #state{gc_next_key = Cont,
                       Now = floor(
                               time_compat:os_system_time(milli_seconds),
                               State),
-                      gc(Key, Table, Policies, Now),
+                      gc(Key, Table, Config, Now),
                       Key
               end,
-    gc_batch(Rows - 1, Policies, State#state{gc_next_key = NewCont}).
+    gc_batch(Rows - 1, Config, State#state{gc_next_key = NewCont}).
+
+gc(Key, Table, Config, Now) ->
+    case lists:member(Table, ?PROC_STATS_TABLES) of
+        true  -> gc_proc(Key, Table, Config, Now);
+        false -> gc_aggr(Key, Table, Config, Now)
+    end.
+
+gc_proc(Key, Table, Config, Now) when Table == connection_stats;
+                                 Table == channel_stats ->
+    Timeout = pget(process_stats_gc_timeout, Config),
+    case ets:lookup(Table, {Key, stats}) of
+        %% Key is already cleared. Skipping
+        []                           -> ok;
+        [{{Key, stats}, _Stats, TS}] -> maybe_gc_process(Key, Table,
+                                                         TS, Now, Timeout)
+    end.
 
-gc(Key, Table, Policies, Now) ->
-    Policy = pget(retention_policy(Table), Policies),
+gc_aggr(Key, Table, Config, Now) ->
+    Policies = pget(policies, Config),
+    Policy   = pget(retention_policy(Table), Policies),
     rabbit_mgmt_stats:gc({Policy, Now}, Table, Key).
 
+maybe_gc_process(Pid, Table, LastStatsTS, Now, Timeout) ->
+    rabbit_log:error("Maybe GC process ~p~n", [{Table, LastStatsTS, Now, Timeout, Pid}]),
+    case Now - LastStatsTS < Timeout of
+        true  -> ok;
+        false ->
+            case process_status(Pid) of
+                %% Process doesn't exist on remote node
+                undefined -> rabbit_log:error("GC process ~p~n", [{Table, Pid}]),
```

Comment:
Same as above, I think it either deserves being a warning, or not being logged at all (might introduce too much log noise).